packages/manager/src/features/CloudPulse/shared/CloudPulseResourcesSelect.tsx

```diff
@@ -73,22 +73,18 @@ export const CloudPulseResourcesSelect = React.memo(
           setSelectedResources([]);
           handleResourcesSelection([]);
         }
-      } else {
+      } else if (selectedResources) {
+        handleResourcesSelection([]);
         setSelectedResources([]);
       }
       // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [resources, region, resourceType, xFilter]);
+    }, [resources]);
 
     return (
       <Autocomplete
         onChange={(_: any, resourceSelections: CloudPulseResources[]) => {
```

Comment:
I'd rather not use it and have it correctly infer than do `any`
```suggestion
        onChange={(e, resourceSelections: CloudPulseResources[]) => {
```